modules/programs/xonsh.nix

```
@@ -0,0 +1,62 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.programs.xonsh;
+  inherit (lib) types mkOption;
+in {
+  options.programs.xonsh = {
+    enable = lib.mkEnableOption "xonsh";
+    enablePromptToolkit =
+      lib.mkEnableOption "advanced prompt based on prompt_toolkit";
+
+    package = lib.mkPackageOption pkgs "xonsh" { };
+    finalPackage = lib.mkOption {
+      type = types.package;
+      internal = true;
+      description = "Package that will actually gat installed";
+    };
+    xonshrc = mkOption {
+      type = types.lines;
+      default = "";
+      description = ''
+        The contents of .xonshrc
+      '';
+    };
+    shellAliases = mkOption {
+      type = with types; attrsOf (listOf str);
+      default = { };
+      example = { ll = [ "ls" "-l" ]; };
+      description = ''
+        An attribute set that maps aliases (the top level attribute names in
+        this option) to commands
+      '';
+    };
+    extraPackages = mkOption {
+      type = with types; functionTo (listOf package);
+      default = _: [ ];
+      defaultText = "_: []";
+      description = ''
+        List of python packages and xontrib to make avaiable
+      '';
+    };
+  };
+  config = lib.mkIf cfg.enable {
+    home = {
+      packages = [ cfg.finalPackage ];
+      file.".xonshrc" = {
+        enable = cfg.xonshrc != "";
+        text = cfg.xonshrc;
+      };
+    };
```

I think it's better to respect XDG.

```suggestion
    home.packages = [ cfg.finalPackage ];
    xdg.configFile."xonsh/rc.xsh" = {
      enable = cfg.xonshrc != "";
      text = cfg.xonshrc;
    };
```

Another option is how https://github.com/nix-community/home-manager/pull/5160 does it: split by files. It's [directly supported by xonsh](https://xon.sh/xonshrc.html#run-control-file).

Which one is better? I don't know, but you and @inmaldrerah should probably join efforts.

Thanks both!